docs/src/pages/components/data-grid/state/state.md

```diff
@@ -0,0 +1,65 @@
+---
+title: Data Grid - State
+---
+
+# Data Grid - State
+
+<p class="description">Initialize and read the state of the data grid.</p>
+
+## Initialize the state
+
+Some state keys can be initialized with the `initialState` prop.
+
+> ⚠️ The `initialState` can only be used to set the initial value of the state, the grid will not react if you change the `initialState` value later on.
+>
+> If you need to fully control some models, use the control props instead (e.g. `prop.filterModel` or `prop.sortModel`).
+> More information on each feature documentation page.
+
+{{"demo": "pages/components/data-grid/state/InitialState.js", "bg": "inline"}}
+
+## Access the state [<span class="plan-pro"></span>](https://mui.com/store/items/material-ui-pro/)
+
+The state is exposed on the `apiRef` object.
+It is strongly advised not to access the state values manually because the structure of the state can change if needs be.
```

Comment:
Maybe that this sounds more descriptive?

```suggestion
It is strongly advised not to access the state values directly because the state itself is not considered a public API and its structure can change.
```